### ISSUE TITLE
Clarification on loss calculation with free bits

### ISSUE THREAD

**Issue by Priya22** [tombosc/exps-s2svae#1]
Going through the code, I notice the following formulation of the free bits loss:
```
kl_mask = (loss_kl > args.target_kl).float() 
loss_rc = loss_rc.sum(-1)  \
loss = loss_rc + kl_mask * kl_weight * loss_kl 
```

(`text_anneal_fb.py`, line 450)

This seems like it sets the KL loss to zero if it's less than the free bits value `fb`, but from what I understand, the value should be clamped everywhere at `min = fb `instead. 

Could you please clarify if I'm missing something? 

Thank you!

**Comment by tombosc** [comment]
Hello @Priya22, sorry for the late answer.

> This seems like it sets the KL loss to zero if it's less than the free bits value fb

Correct

> the value should be clamped everywhere at min = fb instead

Yes, you are right that it does not correspond to the loss p.2 of the paper. However:

- the constant (0 or lambda) do not change the gradients so the optimisation objectives are equivalent.
- nowhere in the paper do we use the value of the loss (sum of max... + reconstruction) as a meaningful measure (I don't think it is)

Please reopen if you still think there is something wrong (and don't hesitate to ping me by email)
